### PR TITLE
Add template translation external ID to msg templating

### PR DESCRIPTION
**Pull request by norkans7**
Contains #1240 and #1239

### CODE REVIEW

File: flows/msg.go [hunk 1]
```diff
@@ -199,13 +201,14 @@ func (tc *TemplatingComponent) Preview(c assets.TemplateComponent) string {
 // MsgTemplating represents any substituted message template that should be applied when sending this message
 type MsgTemplating struct {
 	Template_   *assets.TemplateReference `json:"template"`
+	ExternalID_ string                    `json:"external_id"`
```
Comment (norkans7) [inline flows/msg.go:204]: We need this as for Twilio content template we need to send the template SID to the API and not the template name as we do not for WhatsApp API

Comment (rowanseymour) [reply flows/msg.go:204]: why can't mailroom just resolve that?

Comment (norkans7) [reply flows/msg.go:204]: Okay, I can move that to mailroom

**Comment by norkans7** [comment]
replaced by https://github.com/nyaruka/mailroom/pull/226

**Pull request closed without merging**